packages/apps-config/src/extensions/index.ts

```diff
@@ -37,3 +40,21 @@ export const availableExtensions = known.reduce<Record<Browser, Extension[]>>((a
 
   return available;
 }, { chrome: [], firefox: [] });
+
+// Some extensions do not use `@polkadot/extension-inject` and need to be manually inject into a window context
+
+export function injectExtensions (): Promise<boolean[]> {
+  return Promise.all([
+    initPolkadotSnap
+  ].map((method) => {
+    // Timeout injecting extension
+    return Promise.race([
+      method(),
+      new Promise<false>((resolve) => {
+        setTimeout((): void => {
+          resolve(false);
+        }, 1000 /* 1 sec */);
+      })
+    ]);
+  }));
+}
```

Comment:
1. Why does the return time have to be `Promise<boolean[]>`? After all, `initPolkadotSnap` is just returning a single `Promise<boolean>`.

2. In general, this code could be simplified when using `rxjs` which already is a dependency:

  ```suggestion
  export const injectExtensions = () =>
    firstValueFrom(from(initPolkadotSnap())
      .pipe(
        timeout(1000), // timeout if initPolkadotSnap() doesn't resolve after 1s
        catchError(() => of(false)),
      ));
  ```
  
  (Please note that I haven't actually tested this code)